src/dbnode/storage/bootstrap/bootstrapper/persist.go

```diff
@@ -94,6 +94,14 @@ func PersistBootstrapIndexSegment(
 	retentionOpts := ns.Options().RetentionOptions()
 	nowFn := resultOpts.ClockOptions().NowFn()
 	earliestRetentionTime := retention.FlushTimeStart(retentionOpts, nowFn())
+
+	// If bootstrapping is taking more time than our retention period, we might end up in a situation
+	// when earliestRetentionTime is larger than out block end time. This means that the blocks
+	// got outdated during bootstrap so we just skip building index segments for them.
+	if earliestRetentionTime.After(blockEnd) {
```

Comment:
Shouldn't this also be true when `earliestRetentionTime` equals `blockEnd`?

Reply:
Well, this is the question about block's upper range inclusiveness. If we have a block from `06:00` to `10:00`, do we treat it like `[06:00,10:00]` or `[06:00, 10:00)`? I wasn't sure, so to me it was safer to allow indexing when `earlierRetentionTime` is equals `blockEnd` then to forbid it.

Reply:
`10:00` would belong to the next block, so the end has to be exclusive to avoid overlap: `[06:00, 10:00)`.

Reply:
ok, so if the end is exclusive, then it needs additional equals check.

Reply:
FYI can use `!blockEnd.After(earliestRetentionTime)` for that.

Reply:
Not sure if that's more readable but it is probably more efficient.